rpcserver.go

```diff
@@ -1097,47 +1117,49 @@ func handleCreateRawSSGenTx(s *rpcServer, cmd interface{}, closeChan <-chan stru
 	for i, ssgenPkh := range ssgenPkhs {
 		// Ensure amount is in the valid range for monetary amounts.
 		if ssgenCalcAmts[i] <= 0 || ssgenCalcAmts[i] > dcrutil.MaxAmount {
-			return nil, dcrjson.RPCError{
-				Code:    dcrjson.ErrRPCType,
-				Message: "Invalid amount",
+			return nil, &dcrjson.RPCError{
+				Code: dcrjson.ErrRPCType,
+				Message: fmt.Sprintf("Invalid amount(%v): %v",
+					i, ssgenCalcAmts[i]),
 			}
 		}
 
-		// Create a new script which pays to the provided address specified in
-		// the original ticket tx.
-		var ssgenOutScript []byte
+		// Create a new script which pays to the provided address
+		// specified in the original ticket tx.
+		var ssgenOut []byte
 		switch ssgenPayTypes[i] {
 		case false: // P2PKH
-			ssgenOutScript, err = txscript.PayToSSGenPKHDirect(ssgenPkh)
+			ssgenOut, err = txscript.PayToSSGenPKHDirect(ssgenPkh)
 			if err != nil {
-				errStr := fmt.Sprintf("SSGen out generation failure, pkh bad"+
-					": %v", err)
-				return nil, dcrjson.RPCError{
-					Code:    dcrjson.ErrRPCInvalidParameter,
-					Message: errStr,
+				return nil, &dcrjson.RPCError{
+					Code: dcrjson.ErrRPCInvalidParameter,
+					Message: fmt.Sprintf("Could not "+
+						"generate PKH script: %v", err),
 				}
 			}
 		case true: // P2SH
-			ssgenOutScript, err = txscript.PayToSSGenSHDirect(ssgenPkh)
+			ssgenOut, err = txscript.PayToSSGenSHDirect(ssgenPkh)
 			if err != nil {
-				errStr := fmt.Sprintf("SSGen out generation failure, sh bad"+
-					": %v", err)
-				return nil, dcrjson.RPCError{
-					Code:    dcrjson.ErrRPCInvalidParameter,
-					Message: errStr,
+				return nil, &dcrjson.RPCError{
+					Code: dcrjson.ErrRPCInvalidParameter,
+					Message: fmt.Sprintf("Could not "+
+						"generate SHD script: %v", err),
 				}
 			}
 		}
 
 		// Add the txout to our SSGen tx.
-		txOut := wire.NewTxOut(ssgenCalcAmts[i], ssgenOutScript)
+		txOut := wire.NewTxOut(ssgenCalcAmts[i], ssgenOut)
 		mtx.AddTxOut(txOut)
 	}
 
 	// Check to make sure our SSGen was created correctly.
 	_, err = stake.IsSSGen(mtx)
 	if err != nil {
-		return nil, err
+		return nil, &dcrjson.RPCError{
```

Comment:
There is an `internalRPCError` function that should be used for internal errors.